x/uibc/quota/keeper/msg_server_test.go

```diff
@@ -0,0 +1,142 @@
+//go:build experimental
+// +build experimental
+
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"gotest.tools/v3/assert"
+
+	"github.com/umee-network/umee/v4/x/uibc"
+)
+
+func TestMsgServer_GovUpdateQuota(t *testing.T) {
+	s := initKeeperTestSuite(t)
+	ctx := s.ctx
+
+	tests := []struct {
+		name        string
+		msg         uibc.MsgGovUpdateQuota
+		errExpected bool
+		errMsg      string
+	}{
+		{
+			name: "invlaid authority address in msg",
+			msg: uibc.MsgGovUpdateQuota{
+				Title:       "title",
+				Description: "desc",
+				Authority:   authtypes.NewModuleAddress("govv").String(),
+				Total:       sdk.NewDec(10),
+				PerDenom:    sdk.NewDec(1),
+			},
+			errExpected: true,
+			errMsg:      "expected gov account as only signer for proposal message",
+		},
+		{
+			name: "invalid quota in msg",
+			msg: uibc.MsgGovUpdateQuota{
+				Title:         "title",
+				Description:   "desc",
+				Authority:     authtypes.NewModuleAddress("gov").String(),
+				QuotaDuration: time.Duration(time.Minute * 100),
+				PerDenom:      sdk.NewDec(1000),
+				Total:         sdk.NewDec(100),
+			},
+			errExpected: true,
+			errMsg:      "total quota must be greater than or equal to per_denom quota",
+		},
+		{
+			name: "valid in msg",
+			msg: uibc.MsgGovUpdateQuota{
+				Title:         "title",
+				Description:   "desc",
+				Authority:     authtypes.NewModuleAddress("gov").String(),
+				QuotaDuration: time.Duration(time.Minute * 100),
+				PerDenom:      sdk.NewDec(1000),
+				Total:         sdk.NewDec(10000),
+			},
+			errExpected: false,
+		},
```

Comment:
let's add another msg, with different parmas (quota, etc...) and observe that it changed.